src/yz_kv.erl

```diff
@@ -190,35 +190,54 @@ has_indexes(RemoteNode) ->
         _ -> false
     end.
 
+
+%% @doc Index the data supplied in the Riak Object.
+%% The Riak Object should be a deserialized object (suitable for use with the riak_object
+%% module, or a binary, which has been serialized using riak_object:to_binary/1
+-spec index_binary(bucket(), key(), binary(), atom(), integer()) -> ok.
+index_binary(B, K, Obj, Reason, P) when is_binary(Obj) ->
+    index_internal(
+        Reason, P,
+        fun() -> {B, K} end,
```

Comment:
I don't mind the passing of functions here, but this may work better as named helper functions (for readability) determined by the Obj guard (is_binary or not)